src/main.ts

```diff
@@ -3,57 +3,94 @@ import {createIssueComment, createReview, getExistingIssueComments, getExistingR
+  if (!canCheckCoverity) {
+    warning('Missing Coverity Connect info. Issues will not be checked against the server.')
+  }
+
+  let mergeKeyToIssue = new Map<string, ProjectIssue>()
+  if (canCheckCoverity && coverityIssues && coverityIssues.issues.length > 0) {
+    let covProjectIssues: IIssuesSearchResponse | any = null
+    const apiService = new CoverityApiService(COVERITY_URL, COVERITY_USERNAME, COVERITY_PASSWORD)
+    // TODO page through issues?
+    await apiService
+      .findIssues(COVERITY_PROJECT_NAME, 0, 500)
+      .then(result => (covProjectIssues = result))
+      .catch(error => setFailed(error))
+    info(`Found ${covProjectIssues?.totalRows} potentially matching issues on the server`)
+    mergeKeyToIssue = mapMergeKeys(covProjectIssues)
+  }
+
+  const newReviewComments = []
+  const existingReviewComments = await getExistingReviewComments()
+  const existingIssueComments = await getExistingIssueComments()
+  const diffMap = await getPullRequestDiff().then(getDiffMap)
+
+  for (const issue of coverityIssues.issues) {
+    info(`Found Coverity Issue ${issue.mergeKey} at ${issue.mainEventFilePathname}:${issue.mainEventLineNumber}`)
+
+    const projectIssue = mergeKeyToIssue.get(issue.mergeKey)
+    let ignoredOnServer = false
+    let newOnServer = true
+    if (projectIssue) {
+      ignoredOnServer = projectIssue.action == 'Ignore' || projectIssue.classification in ['False Positive', 'Intentional']
+      newOnServer = projectIssue.firstSnapshotId == projectIssue.lastSnapshotId
+      info(`Issue state on server: ignored=${ignoredOnServer}, new=${newOnServer}`)
     }
 
-    if (newReviewComments.length > 0) {
-      info('Publishing review...')
-      createReview(newReviewComments)
+    const mergeKeyComment = mergeKeyCommentOf(issue)
+    const reviewCommentBody = createMessageFromIssue(issue)
+    const issueCommentBody = createMessageFromIssueWithLineInformation(issue)
+
+    const existingMatchingReviewComment = existingReviewComments
+      .filter(comment => comment.line === issue.mainEventLineNumber)
+      .filter(comment => comment.body.includes(COMMENT_PREFACE))
+      .find(comment => comment.body.includes(mergeKeyComment))
+
+    const existingMatchingIssueComment = existingIssueComments.filter(comment => comment.body?.includes(COMMENT_PREFACE)).find(comment => comment.body?.includes(mergeKeyComment))
+
+    if (existingMatchingReviewComment !== undefined) {
+      info(`Issue already reported in comment ${existingMatchingReviewComment.id}, updating...`)
+      updateExistingReviewComment(existingMatchingReviewComment.id, reviewCommentBody)
+    } else if (existingMatchingIssueComment !== undefined) {
+      info(`Issue already reported in comment ${existingMatchingIssueComment.id}, updating...`)
+      updateExistingIssueComment(existingMatchingIssueComment.id, issueCommentBody)
+    } else if (isInDiff(issue, diffMap)) {
```

Comment:
I think you want to know if it's ignored or not new before you want to know if it's in the diff, this should probably be the second to last else-if